src/Connections/GithubRepo.ts

```diff
@@ -115,46 +204,29 @@ function compareEmojiStrings(e0: string, e1: string, e0Index = 0) {
     return e0.codePointAt(e0Index) === e1.codePointAt(0);
 }
 
-function validateState(state: Record<string, unknown>): GitHubRepoConnectionState {
-    if (typeof state.org !== "string") {
-        throw new ApiError("Expected a 'org' property", ErrCode.BadValue);
-    }
-    if (typeof state.repo !== "string") {
-        throw new ApiError("Expected a 'repo' property", ErrCode.BadValue);
-    }
-    const res: GitHubRepoConnectionState = {
-        org: state.org,
-        repo: state.repo,
-    }
-    if (state.commandPrefix) {
-        if (typeof state.commandPrefix !== "string") {
-            throw new ApiError("Expected 'commandPrefix' to be a string", ErrCode.BadValue);
-        }
-        if (state.commandPrefix.length < 2 || state.commandPrefix.length > 24) {
-            throw new ApiError("Expected 'commandPrefix' to be between 2-24 characters", ErrCode.BadValue);
-        }
-        res.commandPrefix = state.commandPrefix;
-    }
-    if (state.ignoreHooks && Array.isArray(state.ignoreHooks)) {
-        if (state.ignoreHooks?.find((ev) => !AllowedEvents.includes(ev))?.length) {
-            throw new ApiError(`'events' can only contain ${AllowedEvents.join(", ")}`, ErrCode.BadValue);
-        }
-        res.ignoreHooks = state.ignoreHooks;
-    }
-    return res;
-}
-
-
 /**
  * Handles rooms connected to a github repo.
  */
 @Connection
 export class GitHubRepoConnection extends CommandConnection implements IConnection {
+
+	static validateState(state: Record<string, unknown>, isExistingState = false): GitHubRepoConnectionState {
+        const validator = new Ajv().compile(ConnectionStateSchema);
+        if (validator(state)) {
+            // Validate ignoreHooks IF this is an incoming update (we can be less strict for existing state)
+            if (!isExistingState && state.ignoreHooks && !state.ignoreHooks.every(h => AllowedEvents.includes(h))) {
+                throw new ApiError('`ignoreHooks` must only contain allowed values', ErrCode.BadValue);
+            }
+            return state;
+        }
+        throw new ValidatorApiError(validator.errors || []);
```

Comment:
Curious – is there an expected scenario when validation wouldn't pass but `validator.errors` would be empty? The code is sound regardless, I just wonder if it'd happen for anything other than ajv bug.

Reply:
(And if that's valid ajv behaviour, it may be worth teaching ValidatorApiError's constructor to accept falsy `validator.errors` too, so that a future developer doesn't need to remember about this possibility.

Reply:
I think this was just a typing bug yeah. 

>(And if that's valid ajv behaviour, it may be worth teaching ValidatorApiError's constructor to accept falsy validator.errors too, so that a future developer doesn't need to remember about this possibility.

Yep, let's do that.